common.blocks/button/button.bemhtml

```diff
@@ -1,4 +1,11 @@
 block('button')(
+    def()(function() {
+        var isRealButton = (['button', 'input'].indexOf(apply('tag')) >= 0) &&
```

Comment:
@narqo 
For example, we have an attach with such BEMJSON as:

```
    {
        block : 'attach',
        name : 'attach-name',
        button : { block : 'button', mods : { theme : 'islands', size : 'm' } },
        noFileText : 'no file'
    }
```

In this case in `def`-mode:
```
this.ctx.tag || (this.ctx.tag = 'button');
var isRealButton = (['button'].indexOf(this.ctx.tag) >= 0) &&
                (!this.mods.type || this.mods.type === 'submit');
```
So, `isRealButton` will be `true` and `type` attribute will be set as `'button'` (https://github.com/bem/bem-components/blob/issues/1566%40v2/common.blocks/button/button.bemhtml#L39).
After it in [attach__button in `tag` mode](https://github.com/bem/bem-components/blob/issues/1566%40v2/common.blocks/attach/__button/attach__button.bemhtml#L3) we will change tag to `<span>`. As a result, it will be `<span type="button">`. And to fix it now we should add `def` mode to `attach__button` too.